### ISSUE TITLE
Curve leverage lending bot FE revision

### ISSUE THREAD

**Issue by fieldtheory123** [VolumeFi/paloma#1170]
Leverage lending bot FE needs the following changes per user feedback

<img width="231" alt="image" src="https://github.com/VolumeFi/paloma/assets/18745242/740d2644-167b-4356-a311-ffc6a0e2494e">

1. Change "crvUSD Borrow Token" to "The amount of crvUSD token you are borrowing for leverage"
2. Please add the token that is being levered. So, in this case ETH is the token that is being leveraged.
3. Change "Deleverage Percentage" to "Repay Loan Percentage"
4. Remove N slider and make Leverage a Slider that always selects the Max Borrow at a 13% Health Rate
5. Change " Collateral Token" to "Deposit Token" The collateral is ETH

**Comment by taariq** [comment]
@wc117 do we have to make any Vyper contract changes so that we may use ANY collateral in https://crvusd.curve.fi/#/ethereum/markets?

**Comment by verabehr** [comment]
@taariq @fieldtheory123 if a user deposits a non-supported token, do we default to ETH as collateral?

**Comment by taariq** [comment]
@verabehr no. The user chooses the collateral and we swap into that collateral. They must choose what they want to leverage against. We don't want to make that choice for them if Curve allows multiple collateral options.

**Comment by taariq** [comment]
This was a comment from a user I called and who walked through the app.

**Comment by joycehermosilla** [comment]
Just to confirm: Item 5. will not change in the current bot but in the Degen Leverage Bot, right?

**Comment by verabehr** [comment]
@joycehermosilla I'm creating a new ticket for teh degen bot. Only 6 applies to the current Degen bot - cc @jamie088

**Comment by taariq** [comment]
Looks great! Let's GO!

**Comment by jamie088** [comment]
@fieldtheory123 I don't understand about this: `5. Remove N slider and make Leverage a Slider that always selects the Max Borrow at a 13% Health Rate. This Leverage is the new slider.`
My understanding is Health is fixed as 13% and the user can select the Leverage value from Slider. Then we should calculate the Max Borrow amount from Leverage and Health rate. Right?
But if so, I can't do it. Because the curve-js library doesn't support that functionality.

**Comment by fieldtheory123** [comment]
@jamie088 I suppose currently you are calling curve-js' functions to derive health from N and leverage? Can you look at the details of the function and find the formula? If so, you can use that formula and reverse engineer to calculate N from leverage and health.

**Comment by taariq** [comment]
@joycehermosilla & @jamie088: We are going to make a change:

1. The user will only use the slider to select their Leverage
2. The user will not enter in their **crvUSD Borrow Token** This number will always be calculated as 7% Less the MAX crvUSD possible so that the user will never enter this number. It will always be calculated based on the N chosen.
3. We will need the UI to change to show this so that the user understands the max leverage is available for each N.

**Comment by jamie088** [comment]
> @joycehermosilla & @jamie088: We are going to make a change:
> 
> 1. Each Collateral Token will be its own bot. The user no longer chooses Collateral Token.
> 2. The user will only use the slider to select their Leverage
> 3. The user will not enter in their **crvUSD Borrow Token** This number will always be calculated as 15% Less the MAX crvUSD possible so that the user will never enter this number. It will always be calculated based on the N chosen.
> 4. Thus, the user's MAX crvUSD = 85% of available
> 5. N becomes a slider and we remove Bands. The Slider is from LEFT to RIGHT OR LOW TO HIGH. Thus N=50 will start left and N=4 to the right, but the user never sees "bands" concept. They just use the slider
> 6. Input default health repayment levels is 50% of the health at the chosen leverage. User can update this
> 7. % to repay will be default 50%. User can update this.

@taariq My question for your requirements.
1. If the user no longer chooses Collateral Token in UI, which token should be set to as a Collateral Token?
2. Then, the UI has two sliders like leverage and N. Right?

**Comment by taariq** [comment]
@jamie088: 
1. Each collateral token will have its own Bot Front-end for Leverage. @joycehermosilla will prepare 5 front-end designs featuring the tokens for each bot
2. The UI has only 1 SLIDER. That Slider is for leverage. It will be the N slider, but in the reverse direction. The user won't see bands. We'll just do 50(Left/Minimum Leverage) to 4(Right/Max Leverage)

**Comment by jamie088** [comment]
https://www.figma.com/file/6DYR6QFkucGd1bsOILr7V0/Paloma-Bot?type=design&node-id=1653-464200&mode=design&t=mJDEjEDCkufA7tc6-4

**Comment by taariq** [comment]
Metamask error:

(anonymous) @ lockdown-install.js:1
Show 1 more frame
Show less
lockdown-install.js:1 null
(anonymous) @ lockdown-install.js:1

**Comment by taariq** [comment]
![Image](https://github.com/VolumeFi/paloma/assets/701864/e526c164-034e-475e-bb34-ef02c4356d68)

**Comment by taariq** [comment]
Change: **Borrow Rate --> Loan interest rate**
